### PR TITLE
fix: filter out lenses post query as opposed to in ODS query

### PR DESCRIPTION
**Pull request by bryanjknight**
This fix no longer filters out lens data as part of the FHIR query but rather filters out the lenses post query. This reduce query times from 10s to 100ms

### CODE REVIEW

File: src/components/content/conditions/helpers/history.tsx [hunk 1]
```diff
@@ -15,9 +17,35 @@ export function useConditionHistory(condition: ConditionModel) {
...
+  if (!metaTags) {
+    return true;
+  }
+
+  const lenTags = metaTags.filter((tag) => 
+    isChronicConditionLens(tag.system, tag.code) ||
+    isSummaryConditionLens(tag.system, tag.code)
+  )
+
+  const hasLensTags = lenTags.length > 0
```
Comment (pmadria3) [inline src/components/content/conditions/helpers/history.tsx]:
```suggestion
  const lenTags = metaTags.filter((tag) => 
    isChronicConditionLens(tag.system, tag.code) ||
    isSummaryConditionLens(tag.system, tag.code)
  ).length > 0
```

Nit: you can combine the .length and then just do the not of that and return it

Comment (bryanjknight) [reply src/components/content/conditions/helpers/history.tsx]: yeah, I went with your suggestion but also like decoupling what's returned from the business logic. If we wanted to do additional filtering, we'd just have to add another boolean to the return value. Certainly YAGNI though so game for taking the suggestion

**Review by pmadria3 — DISMISSED** [review]
LGTM - not sure if Eric is also reviewing or not

File: src/fhir/search-helpers.ts [hunk 2]
```diff
@@ -199,16 +199,24 @@ export async function searchSummaryRecords<T extends ResourceTypeString>(
...
 
-// Like searchAllRecords, but filters out lens resources.
+// Like searchAllRecords, but filters out lens resources. If postQueryFilter
+// is specified, exclusion will happen client side, not server side
 export async function searchCommonRecords<T extends ResourceTypeString>(
   resourceType: T,
   requestContext: CTWRequestContext,
-  searchParams?: SearchParams
+  searchParams?: SearchParams,
+  postQueryFilter?: (r: ResourceType<T>) => boolean
```
Comment (eric-gunther) [inline src/fhir/search-helpers.ts:208]: I don't think `postQueryFilter` should be an input here, `searchCommonRecords` should simply filter out lens resources as part of its default behavior with no other options.

File: src/components/content/conditions/helpers/history.tsx [hunk 3]
```diff
@@ -15,9 +17,33 @@ export function useConditionHistory(condition: ConditionModel) {
...
+    return true;
+  }
+
+  const hasLensTags =
+    metaTags.filter(
+      (tag) =>
+        isChronicConditionLens(tag.system, tag.code) || isSummaryConditionLens(tag.system, tag.code)
+    ).length > 0;
+
+  return !hasLensTags;
```
Comment (eric-gunther) [inline src/components/content/conditions/helpers/history.tsx:44]: Per the other comment, I think all this logic doesn't belong in history.tsx, I think it belongs inside `searchCommonRecords`.

**Review by eric-gunther — CHANGES_REQUESTED** [review]

File: src/fhir/search-helpers.ts (hunk 2 above)
Comment (bryanjknight) [reply src/fhir/search-helpers.ts:208]: Gotcha, so just filter out any lens (ActiveMedication, ChronicCondition, Summary, etc)

File: src/fhir/search-helpers.ts [hunk 4]
```diff
@@ -199,16 +199,33 @@ export async function searchSummaryRecords<T extends ResourceTypeString>(
   return records;
 }
 
-// Like searchAllRecords, but filters out lens resources.
+// Like searchAllRecords, but filters out lens resources. If postQueryFilter
```
Comment (eric-gunther) [inline src/fhir/search-helpers.ts:202]:
```suggestion
// Like searchAllRecords, but filters out lens resources.
```

File: src/fhir/search-helpers.ts [hunk 5]
```diff
@@ -199,16 +199,33 @@ export async function searchSummaryRecords<T extends ResourceTypeString>(
...
+// is specified, exclusion will happen client side, not server side
```
Comment (eric-gunther) [inline src/fhir/search-helpers.ts:203]:
```suggestion
```

**Review by eric-gunther — APPROVED** [review]

**Pull request merged**